spring-cloud-stream-binder-kinesis/src/main/java/org/springframework/cloud/stream/binder/kinesis/config/KinesisBinderConfiguration.java

```diff
@@ -67,14 +72,44 @@ public KinesisStreamProvisioner provisioningProvider(AmazonKinesisAsync amazonKi
 	KinesisMessageChannelBinder kinesisMessageChannelBinder(
 			AmazonKinesisAsync amazonKinesis,
 			KinesisStreamProvisioner provisioningProvider,
-			Codec codec) {
+			Codec codec, MetadataStore kinesisCheckpointStore) {
 
 		KinesisMessageChannelBinder kinesisMessageChannelBinder =
 				new KinesisMessageChannelBinder(amazonKinesis, this.configurationProperties, provisioningProvider);
 		kinesisMessageChannelBinder.setCodec(codec);
+		kinesisMessageChannelBinder.setMetaDataStore(kinesisCheckpointStore);
 
 		return kinesisMessageChannelBinder;
 	}
 
+	
+	@Bean
+	public AmazonDynamoDBAsync dynamoDB(AWSCredentialsProvider awsCredentialsProvider,
+			RegionProvider regionProvider) {
+		
+		AmazonDynamoDBAsync dynamoDB = AmazonDynamoDBAsyncClientBuilder.standard()
+				.withCredentials(awsCredentialsProvider)
+				.withRegion(regionProvider.getRegion().getName())
+				.build();
+
+		return dynamoDB;
+	}
+	
+	
+	@Bean
+	MetadataStore kinesisCheckpointStore(AmazonDynamoDBAsync dynamoDB) {
```

Comment:
In the issue I requested to mark this bean with the `@ConditionalOnMissingBean` to let end-user to override it.
For example we have to do that in our integration tests against Kinesalite. The Dynalite is started on the same port by default. I'm pretty sure the `SimpleMetadataStore` is enough for tests. So, such a bean must be declared there overriding this one for the `DynamoDbMetaDataStore`